core/controller/src/main/scala/whisk/core/controller/actions/PrimitiveActions.scala

```diff
@@ -152,19 +150,12 @@ protected[actions] trait PrimitiveActions {
                 pollDbForResult(docid, activationId, promise)
         }
 
-        val response = promise.future withTimeout (totalWaitTime, new BlockingInvokeTimeout(activationId))
-
-        response onComplete {
-            case Success(_) =>
-                // Duration of the blocking activation in Controller.
-                // We use the start time of the tid instead of a startMarker to avoid passing the start marker around.
-                transid.finished(this, StartMarker(transid.meta.start, LoggingMarkers.CONTROLLER_ACTIVATION_BLOCKING))
-            case Failure(t) =>
-                // short circuits db polling
-                promise.tryFailure(t)
-        }
-
-        response // will either complete with activation or fail with timeout
+        // Will either complete with activation or timeout
+        promise.future
+            .withTimeout(totalWaitTime, new BlockingInvokeTimeout(activationId))
+            .andThen {
```

Comment:
Nope. `tryFailure` returns a `Boolean` which'll alter the return-type. `andThen` just passes the original future through and only applies the sideeffect.